File: C++/019_Remove Nth Node From End of List.cpp
class Solution {
public:
    ListNode *removeNthFromEnd(ListNode *head, int n) {
        ListNode *fr = head;
        while (fr != NULL && n--)
            fr = fr->next;
        if (fr != NULL) {
            ListNode *se = head;
            while (fr->next != NULL) {
                fr = fr->next;
                se = se->next;
            }
            ListNode *tmp = se->next;
            se->next = tmp->next;
        }
        else if (n == 0)
            return head->next;
        return head;
    }
};

File: Python/019_Remove Nth Node From End of List.py
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        fast = head
        while fast is not None and n > 0:
            fast = fast.next
            n -= 1
        slow, pre = head, None
        while fast is not None:
            pre = slow
            slow = slow.next
            fast = fast.next
        if pre is None:
            return head.next
        pre.next = slow.next
        return head